include/h2o/dsr.h

```diff
@@ -35,24 +35,35 @@
  */
 typedef struct st_h2o_dsr_req_t {
     /**
-     * The transport protocol for which DSR is performed (e.g., quic-29). This value indicates how the packet should be protected,
-     * aside from the cipher-suite being used.
+     * HTTP version of front-end connection
      */
-    uint32_t quic_version;
-    /**
-     * The cipher-suite being used, as the ID registered to the TLS Cipher Suites registry.
-     */
-    uint16_t cipher;
-    /**
-     * address from which QUIC packets should be sent
-     */
-    quicly_address_t address;
+    int http_version;
+    struct {
+        /*
+         * We use struct instead of union here to make it easier when parsing a serialized string.
+         */
+        struct {
+            /**
+             * The transport protocol for which DSR is performed (e.g., quic-29). This value indicates how the packet should be
+             * protected, aside from the cipher-suite being used.
+             */
+            uint32_t version;
+            /**
+             * The cipher-suite being used, as the ID registered to the TLS Cipher Suites registry.
+             */
+            uint16_t cipher;
+            /**
+             * address from which QUIC packets should be sent
+             */
+            quicly_address_t address;
+        } quic;
+    } transport;
```

Comment:
Can it be just `quic` rather than `transport.quic`?

Reply:
We already have code using unnamed union; it is a  C99 feature. Maybe you got confused with it being introduced to C++ in C++11?

Reply:
Addressed in [7137cba](https://github.com/h2o/h2o/pull/3124/commits/7137cba451fc6fa7728b65382eb3445b1ea4afe2). Ended up with `h3.quic` instead of `transport.quic` as we discussed offline. Please take a look and tell me what you think.